### ISSUE TITLE
feature request: quick color selection using number keys (1-5) while using the drawing tool

### ISSUE THREAD

**Issue by backpropogated** [excalidraw/excalidraw#8726]
i would like to request a feature that allows users to quickly select colors using the number keys (1-5) while using the drawing tool in excalidraw. 

**current workflow issue**:
- the current flow of changing tools is messy. for instance, when i enter color select mode using 's', but then try to select a color via the 'asdfg' hotkeys, it changes tools after 2-3 clicks on the color. this disrupts my drawing flow and makes it frustrating to switch colors efficiently.

**use case**:
- implementing a quick color selection method would enhance my workflow, especially in zen mode where i prefer a more focused experience.

**proposed solution**:
- allow users to assign specific colors to the number keys (1-5) for quick access while drawing, without changing the drawing tool.
- this way, i can easily change colors without interrupting my drawing flow.

thank you for considering this request!

**Comment by priyankuhazarika** [comment]
I want to try this, can you assign this to me @Mrazator

**Comment by dwelle** [comment]
We can't use `1-5` for colors as they're mapped to the toolbar. But I was thinking about about `1-5` _when_ the color picker is open.

**Comment by dwelle** [comment]
@backpropogated  Also, can you elaborate on this:

> but then try to select a color via the 'asdfg' hotkeys, it changes tools after 2-3 clicks on the color. 

What do you mean but it changing tools? Is it while the color picker is still open?

---

as for

> allow users to assign specific colors to the number keys (1-5) for quick access while drawing, without changing the drawing tool.

I may have misunderstood. If you mean allowing to remap shortcuts, then that we can do, but will need to wait for a larger keyboard handling rewrite.

**Comment by KozielGPC** [comment]
Hello everyone! I'm a computer science student and I'm enrolled in a class named "Open Source Development", where we have to contribue to open source projects. I was amazed by this project and the idea, and I'm willing to contribute for it.
May I work on this issue?

**Comment by Mrazator** [comment]
@KozielGPC sure, this sounds ok:

> But I was thinking about about 1-5 when the color picker is open.

**Comment by KozielGPC** [comment]
@Mrazator Sure, thank you!

Just for clarify the things, the 1-5 shortcuts would be for these colors?
![Image](https://github.com/user-attachments/assets/acf26532-5b50-40c6-8bf0-3c50e330afbb)

Or these colors? 
![Image](https://github.com/user-attachments/assets/9f58b453-966e-430b-bd61-7dd73118d7a3)

**Comment by Mrazator** [comment]
The first ones, and should be shown only when the color picker is opened. The shades (Tons) should stay as is.

**Comment by Mrazator** [comment]
Actually just to clarify, there is stroke (`s`) and background (`g`) and it makes sense allowing the shortcuts for both.

![Image](https://github.com/user-attachments/assets/4db061b5-2039-4ff1-9a6e-613afaae05a3)

**Comment by KozielGPC** [comment]
But would it be different shortcuts for both? Like 1-5 for Stroke and 6-0 for Background?

**Comment by Mrazator** [comment]
No, since the context is different, meaning if the stroke color picker is opened, 1-5 changes the main stroke color. If the background color picker is opened, 1-5 changes the main background color.

**Comment by KozielGPC** [comment]
Ahhh I see, makes sense. Okay!
